lib/portus/navalia_client.rb

```diff
@@ -0,0 +1,66 @@
+module Portus
+  class NavaliaClient
+    include HttpHelpers
+    def initialize(host)
+      @host = host
+      @base_url = "http://#{@host}"
+    end
+
+    # Returns whether the registry is reachable with the given credentials ori
+    # not.
+    def reachable?
+      begin
+        !perform_request("/ping").nil?
+      rescue
+        false
+      end
+    end
+
+    # Returns the status of the builds
+    def status(ids)
+      body = { "ids" => ids }
+      remote_call("build", body, :get)
+    end
+
+    # Delete builds
+    def delete(ids)
+      body = { "ids" => ids }
+      remote_call("build", body, :delete)
+    end
+
+    # Trigger a build
+    def build(url, registry, image_id)
+      body = {
+        "url" => url,
+        "push" =>
+        {
+          "id" => "registry",
+          "hostname" => registry,
+          "image" => image_id}
+      }
+      remote_call("build", body, :post)
+    end
+
+    protected
+
+    def remote_call(action, body, type)
```

Comment:
Instead of doing this, why don't you re-use the `perform_request` method as provided by the module `HttpHelpers` which you're including ? I guess that it's because you want to add extra headers, so why don't you extend the `perform_request` method so it can be provided an optional hash of headers (while being clever with the `Authorization` header).

I'm saying this, because:

- The implementation already takes care of stuff like timeouts.
- We'd avoid code duplication.